core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -27,7 +27,7 @@ trait RTS {
 
   final def unsafeRunAsync[E, A](io: IO[E, A])(k: Callback[E, A]): Unit = {
     val context = new FiberContext[E, A](this, defaultHandler)
-    context.evaluate(io)
+    context.evaluate(this, io)
```

Comment:
The runtime system is already passed at construction time (see the line just above), so you don't need to thread it everywhere.